### PR TITLE
(DOCSP-29739): C++: Add documentation for Set data type

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-29739

### Staged Changes

- [Supported Types](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-29739/sdk/cpp/model-data/supported-types/#set)
- [CRUD/Create](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-29739/sdk/cpp/crud/create/#create-an-object-with-a-set-property)
- [CRUD/Read](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-29739/sdk/cpp/crud/read/#read-a-set-property)
- [CRUD/Update](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-29739/sdk/cpp/crud/update/#update-a-set-property)
- [CRUD/Delete](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-29739/sdk/cpp/crud/delete/#delete-set-values)
- [React to Changes](https://preview-mongodbdacharyc.gatsbyjs.io/realm/DOCSP-29739/sdk/cpp/react-to-changes/#register-a-collection-change-listener)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Review by cbush — APPROVED** [review]
Seems fine!

File: source/sdk/cpp/model-data/relationships.txt [hunk 1]
```diff
@@ -46,10 +46,11 @@ A **to-many** relationship means that an object relates to more than one
 other object. In Realm, a to-many relationship is a list of 
 references to other objects. For example, a person might have many dogs.
 
-Realm represents a to-many relationship between two Realm
-types as a list. Lists are mutable: within a write transaction, you can 
-add and remove elements to and from a list. Lists are not associated with 
-a query and are declared as a property of the object model.
+You can represents a to-many relationship between two Realm
+types as a list or a set. Lists and sets are mutable: within a write
```
Comment (leemaguire) [inline source/sdk/cpp/model-data/relationships.txt:50]: `map` should be mentioned here too

File: source/sdk/cpp/react-to-changes.txt [hunk 2]
```diff
@@ -65,7 +65,7 @@ Register a Collection Change Listener
 -------------------------------------
 
 You can register a notification handler on a collection. A collection is
-a list property that can contain any :ref:`supported data type 
+a list or set property that can contain any :ref:`supported data type
```
Comment (leemaguire) [inline source/sdk/cpp/react-to-changes.txt:68]: `map` should be mentioned here too as it supports change notifications

**Review by leemaguire — COMMENTED** [review]
Looking good, just some minor comments

**Pull request merged**